### PR TITLE
Changes for story2, add form/sql insert section with validation of fo…

### PR DESCRIPTION
**Pull request by gerryg12**
…rm fields

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -1,31 +1,81 @@
...
+
+    if (isset($_POST['commissioned'])) {
+        $inputtedCommissioned = $_POST['commissioned'];
+        if (strlen($inputtedCommissioned) == 0) {
+            echo "no date inputted";
+        } elseif (strlen($inputtedCommissioned) > 12) {
+            echo "Date is too long";
+                } elseif (!preg_match("/^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])$/",$inputtedCommissioned)) {
+                    echo "Date format invalid";
+        }
```
Comment (ashleycoles) [inline index.php]: Check this bit of indentation, it's indented too far

File: index.php [hunk 2]
```diff
@@ -1,31 +1,81 @@
...
+    }
+
+    // $iputtedModel = $_POST['model'];
+    // $inputtedCommissioned = $_POST['commissioned'];
+
+    $insertMonitor = new MonitorModels($db);
+    $insertMonitor->insertNewMonitor($inputtedMake, $inputtedModel, $inputtedCommissioned);
```
Comment (ashleycoles) [inline index.php]: These two lines shoud now only run if there were no validation issues above. What you could do is create a variable before the validation if statements $valid = true - and then set it to false within the if statments whenever you find an issue.

You can then use that in an if statement to only run model and insertNewMonitor if valid is still true

File: index.php [hunk 3]
```diff
@@ -1,31 +1,81 @@
...
 
-foreach ($monitors as $monitor) {
-    echo "$monitor->make\n"."$monitor->model\n".$monitor->commissioned."<br>";
-}
+    <form method="POST">
+        <label for="make">Make:</label>
+        <input type="text" id="make" name="make">
+        <label for="model">Model:</label>
+        <input type="text" id="model" name="model">
+        <label for="commissioned">Commissioned:</label>
```
Comment (ashleycoles) [inline index.php]: You should add some more info here to tell users what format the date needs to be in

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: index.php (hunk 1 above)
Comment (gerryg12) [reply index.php]: Done

File: index.php (hunk 3 above)
Comment (gerryg12) [reply index.php]: Added date format to label

File: index.php (hunk 2 above)
Comment (gerryg12) [reply index.php]: Additional if statement added to prevent unnecessary code running.

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**